tests/closures.test.ts

```diff
@@ -162,4 +162,155 @@ describe("Closures group's test cases on the proposal", () => {
   f(10)()
   `;
   assert("10. An escaping function calls its non-escaping sibling", src, PyInt(11));
+
+  src = `
+  def concat(items: [bool], stuff: [bool]) -> [bool]:
+    concatted : [bool] = None
+    concatted = items + stuff
+    return concatted
+
+  items : [bool] = None
+  stuff : [bool] = None
+  concatted : [bool] = None
+  items = [True, True, False]
+  stuff = [False, True]
+  concatted = concat(items, stuff)
+`;
+  asserts("11. A function with list created", [
+    [src, PyNone()],
+    ["concatted[0]", PyBool(true)],
+    ["concatted[1]", PyBool(true)],
+    ["concatted[2]", PyBool(false)],
+    ["concatted[3]", PyBool(false)],
+    ["concatted[4]", PyBool(true)],
+  ]);
+
+  src = `
+  def f(x: [int]) -> [int]:
+    def inc() -> [int]:
+      return x + [1]
+    return inc()
+  x : [int] = None
+  x = f([1])
+  x = f(x)
+  x = f(x)
+  `;
+  asserts("12. A nested function with list created", [
+    [src, PyNone()],
+    ["x[0]", PyInt(1)],
+    ["x[1]", PyInt(1)],
+    ["x[2]", PyInt(1)],
+  ]);
+
+  describe.skip("nested func - list", function () {
```

Comment:
We added some test cases about using other group's features inside functions. Note that there are 2 cases that are marked as skipped now: a case for list and a case for for.
For the list case, we can not make it run, probably because that we are not that familiar with the strategies used by the list group. The program raised an exception about memory outage but we were not managed to find anything weird on the flattened AST.